testing/test_assertrewrite.py

```diff
@@ -147,17 +147,41 @@ def f():
         def f():
             assert sys == 42
 
-        assert getmsg(f, {"sys": sys}) == "assert sys == 42"
+        verbose = request.config.getoption("verbose")
+        msg = getmsg(f, {"sys": sys})
+        if verbose:
```

Comment:
I'm not sure we should be having different checks based on the verbosity the user used in the command-line. This might cause confusion, for example if I execute my tests manually without verbosity and it works fine, but breaks on CI.

Is there a way for us to reset the verbosity in `testdir`, so that it will execute with the standard verbosity by default, unless explicitly told otherwise in the `runpytest` call?

Reply:
> Is there a way for us to reset the verbosity in `testdir`, so that it will execute with the standard verbosity by default, unless explicitly told otherwise in the `runpytest` call?

Not sure - after all the PYTEST_ADDOPTS are unset already.
This appears due to the outer pytest already rewriting the code to be verbose.

I think it is OK to have a few conditions like this, but it should not be the common practice, of course - assert rewriting is kind of special in that sense.

Reply:
> Not sure - after all the PYTEST_ADDOPTS are unset already.
This appears due to the outer pytest already rewriting the code to be verbose.

I meant by running `pytest` manually... I do it frequently when developing.

> assert rewriting is kind of special in that sense.

Hmmm right, now that I look closer I see that those are inline tests, not run by `pytester`...

Well I guess it is fine to keep that to only a few tests if we don't find a way around it, I believe.

Reply:
> Is there a way for us to reset the verbosity in `testdir`, so that it will
> execute with the standard verbosity by default, unless explicitly told
> otherwise in the `runpytest` call?

> > Not sure - after all the PYTEST_ADDOPTS are unset already.
> > This appears due to the outer pytest already rewriting the code to be verbose.
> 
> I meant by running `pytest` manually... I do it frequently when developing.

The inner runpytestcall?

There are only args to control it, so `testdir.runpytest("--verbosity=1")` ?

Reply:
btw: just found this https://github.com/blueyed/pytest/blob/f43c54e8b2a6d6975f74f7374d87f6b3fe42cfad/testing/test_assertion.py#L20-L29